scipy/stats/_continuous_distns.py

```diff
@@ -3313,6 +3314,312 @@ def _ppf(self, q, k, s):
+
+        def f2(x, h, k):
+            '''cdf = (1.0 - h*exp(-x))**(1.0/h)
+               logcdf = ...
+            '''
+            return (1.0/h)*special.log1p(-h*exp(-x))
+
+        def f3(x, h, k):
+            '''cdf = exp(-exp(-x))
+               logcdf = ...
+            '''
+            return -exp(-x)
+
+        return _lazyselect(condlist,
+                           [f0, f1, f2, f3],
+                           [x, h, k],
+                           default=nan)
+
+    def _ppf(self, q, h, k):
+        condlist = [np.logical_and(h != 0, k != 0),
+                    np.logical_and(h == 0, k != 0),
+                    np.logical_and(h != 0, k == 0),
+                    np.logical_and(h == 0, k == 0)]
+
+        def f0(q, h, k):
+            return 1.0/k*(1.0 - ((1.0 - (q**h))/h)**k)
+
+        def f1(q, h, k):
+            return 1.0/k*(1.0 - (-log(q))**k)
+
+        def f2(q, h, k):
+            '''ppf = -log((1.0 - (q**h))/h)
+            '''
+            return -special.log1p(-(q**h)) + log(h)
+
+        def f3(q, h, k):
+            return -log(-log(q))
+
+        return _lazyselect(condlist,
+                           [f0, f1, f2, f3],
+                           [q, h, k],
+                           default=nan)
+
+    def _stats(self, h, k):
+        if h >= 0 and k >= 0:
+            maxr = 5
+        elif h < 0 and k >= 0:
+            maxr = int(-1.0/h*k)
+        elif k < 0:
+            maxr = int(-1.0/k)
+        else:
+            maxr = 5
+
+        outputs = [None if r < maxr else nan for r in range(1, 5)]
+        return outputs[:]
+kappa4 = kappa4_gen(name='kappa4')
+
+
+class kappa3_gen(rv_continuous):
+    """Kappa 3 parameter distribution.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for `kappa` is::
+
+        kappa3.pdf(x, a) =
+            a*[a + x**a]**(-(a + 1)/a),     for ``x > 0``
+            0.0,                            for ``x <= 0``
+
+    `kappa3` takes ``a`` as a shape parameter and ``a > 0``.
+
+    References
+    ----------
+    P.W. Mielke and E.S. Johnson, "Three-Parameter Kappa Distribution Maximum
+    Likelihood and Likelihood Ratio Tests", Methods in Weather Research,
+    701-707, (September, 1973),
+    http://docs.lib.noaa.gov/rescue/mwr/101/mwr-101-09-0701.pdf
+
+    B. Kumphon, "Maximum Entropy and Maximum Likelihood Estimation for the
+    Three-Parameter Kappa Distribution", Open Journal of Statistics, vol 2,
+    415-419 (2012)
+    http://file.scirp.org/pdf/OJS20120400011_95789012.pdf
+
+    %(after_notes)s
+
+    %(example)s
+
+    """
+    def _argcheck(self, a):
+        return a > 0
+
+    def _pdf(self, x, a):
+        x = asarray(x)
+        ret = np.zeros(shape(x), dtype=x.dtype)
+        ret[x > 0] = a*(a + x**a)**(-1.0/a-1)
+        return ret
+
+    def _cdf(self, x, a):
+        x = asarray(x)
+        ret = np.zeros(shape(x), dtype=x.dtype)
+        ret[x > 0] = x*(a + x**a)**(-1.0/a)
```

Comment:
I believe this can still be simplified to just `return x * (a + x**2) ** (-1.0/a)`. Everything else is taken care of automagically by the framework. (possibly with the help of `_open_support_mask`) But this can be cleaned up separately.